ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -695,30 +695,49 @@ - (UIImageView *)findHairlineViewUnder:(UIView *)view {
 - (NSArray *)managedResults {
     NSMutableArray *results = [NSMutableArray new];
     
-    [_managedStepIdentifiers enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
-        NSString *identifier = obj;
-        ORKResult *result = _managedResults[identifier];
-        NSAssert(result, @"Result should not be nil for identifier %@", identifier);
+    [_managedStepIdentifiers enumerateObjectsUsingBlock:^(NSString *identifier, NSUInteger idx, BOOL *stop) {
+        id <NSCopying> key = [self uniqueManagedKey:identifier index:idx];
+        ORKResult *result = _managedResults[key];
+        NSAssert2(result, @"Result should not be nil for identifier %@ with key %@", identifier, key);
         [results addObject:result];
     }];
     
     return [results copy];
 }
 
-- (void)setManagedResult:(id)result forKey:(id <NSCopying>)aKey {
+- (void)setManagedResult:(ORKStepResult *)result forKey:(NSString *)aKey {
     if (aKey == nil) {
         return;
     }
     
-    if (result == nil || NO == [result isKindOfClass:[ORKResult class]]) {
+    if (result == nil || NO == [result isKindOfClass:[ORKStepResult class]]) {
         @throw [NSException exceptionWithName:NSGenericException reason:[NSString stringWithFormat: @"Expect result object to be ORKResult type and not nil: {%@ : %@}", aKey, result] userInfo:nil];
```

Comment:
Perhaps change the exception message to something like `'result' shouldn't be nil and should be of type 'ORKStepResult': {%@ : %@}`? :-)